src/prefect/engine/signals.py

```diff
@@ -4,10 +4,38 @@
 in states.
 """
 
+from typing import Type
+
 from prefect.engine import state
 from prefect.utilities.exceptions import PrefectError
 
 
+def signal_from_state(state: state.State) -> Type["PrefectStateSignal"]:
+    """
+    Given a state instance, returns the corresponding Signal type.
+
+    Args:
+        - state (State): an instance of a Prefect State
+
+    Returns:
+        - PrefectStateSignal: the Prefect Signal corresponding to
+            the provided state
+
+    Raises:
+        - KeyError: if no signal matches the provided state
+    """
+    unprocessed = set(PrefectStateSignal.__subclasses__())
+    signals = dict()
+    while unprocessed:
+        sig = unprocessed.pop()
+        signals[sig._state_cls.__name__] = sig
+        unprocessed = unprocessed.union(sig.__subclasses__())
+    try:
+        return signals[type(state).__name__]
+    except KeyError:
+        raise KeyError(f"No signal matches the provided state: {state}") from None
```

Comment:
Minor nit, feel free to ignore - this feels like it should be a `ValueError` instead, since the `state` value isn't a valid value. `KeyError` is generally for user-facing code that is doing an actual indexing operation.

Reply:
👍  works for me, will update